pkg/util/utils.go

```diff
@@ -79,18 +80,20 @@ func ParseChanges(option string) (key string, vals []string, err error) {
 	// 3. key ["value","value1"]
 	if strings.Contains(option, " ") {
 		// This handles 2 & 3 conditions.
-		var val string
-		tokens := strings.SplitAfterN(option, " ", 2)
+		tokens := strings.SplitN(option, " ", 2)
 		if len(tokens) < 2 {
 			return "", []string{}, fmt.Errorf("invalid key value %s", option)
 		}
 		key = strings.Trim(tokens[0], " ") // Need to trim whitespace part of delimiter.
-		val = tokens[1]
-		if strings.Contains(tokens[1], "[") && strings.Contains(tokens[1], "]") {
-			//Trim '[',']' if exist.
-			val = strings.TrimLeft(strings.TrimRight(tokens[1], "]"), "[")
+
+		if strings.Contains(tokens[1], "[") || strings.Contains(tokens[1], "]") {
+			// Try a JSON unmarshal to catch case 3
+			if err := json.Unmarshal([]byte(tokens[1]), &vals); err != nil {
+				return "", []string{}, errors.Wrapf(err, "cannot unmarshal %s as JSON array", tokens[1])
```

Comment:
should we fail out here if the umarshall fails or just stuff tokens[1] into vals and continue on after noting the unmarshal failure in a debug message?

Reply:
I was debating that, but the tests fail out on those situations, so long as `[` or `]` are present, so I kept that behaviour for now.